Sources/COWMacros/SyntaxExtensions.swift

```diff
@@ -119,6 +119,14 @@ extension StructDeclSyntax {
   
 }
 
+extension FunctionSignatureSyntax {
+  
+  var hasThrows: Bool {
```

Comment:
Explicit `internal` access control is recommended. This convention is learned from the Swift Standard Library.

Reply:
If you would like to make an explicit coding standard/convention that this repo should follow, I am fine with that. But I strongly disagree that we should follow the coding convention of standard library. I cannot imagine how terrible it would look like if a C++ project followed the coding convention of libstdcxx.

Also, the official Swift Book explicitly says:

> As a result, in many cases you don’t need to specify an explicit access level in your code.

... which I would like to interpret in this way: implicit `internal` is preferred.

https://docs.swift.org/swift-book/documentation/the-swift-programming-language/accesscontrol/#Default-Access-Levels

Reply:
The analogue to following the code style of libstdcxx is not appropriate here. The readability of Swift Standard Library is much better than libstdcxx's, comparatively.

My point is that "explicit access control" leads to "coders enforcing clarity regarding the scope of each part of the program." When we make "explicit access control" a requirement, this enforcement can be adopted by anyone who implements it.

This might not be significant to those whom Swift is mainly designed for -- the developers who create apps, or say, executables. However, for framework authors, this contributes to a better designed foundation that a lot of people can leverage.

An example could be: a programmer that does not get used to regard the scope of each part of the program uses an implicit-access-control modified `internal` top-level global variable v.s. a programmer get used to that uses an explicit-access-control modified `internal` top-level global variable. The previous one is more likely to be misused.

Reply:
First of all, I am sincerely sorry that it is me who first deviated the discussion that I said I prefer the implicit one.

Let me put it straight: this is your project, and I will follow whatever you suggest, AS LONG AS IT MAKES SENSE. I am perfectly fine that you prefer verbosity, but "following the style of the standard library nobrainly" is definitely a terrible idea. It has nothing to do with the readability of the library itself, it is that the standard library and other 3rd party code serve COMPLETELY different purposes that they MUST be written somewhat differently.

And I have zero interests discussing coding convention here because this not only waste my time, it also waste YOURS. This is a project focusing on productivity, rather than a draft that some sort of language workgroup people debate endlessly. My sincere suggestion is: to make a coding standard guideline in this repo, or to build a pre-merge pipeline that does the checks and outright blocks any violations, rather than wasting more time on this.

Reply:
First, I apologize that I didn't mention this code style in the README.md. The conventions should be mentioned when newcomers join the project. I would like to add this to README.md in time.

However, allow me to explain myself. I didn't mean to follow the Swift Standard Library without further thinking. I only talked about what I learned from it.

In short, my point is that explicit access control results from a clear understanding of the scope of each declaration of the program. It prevents people from abusing variables, structs, classes or any other declarations.

One example is that you modified `private let defaultStorageName: TokenSyntax` into `let defaultStorageName: TokenSyntax` to use it in `COWExpansionFactory.swift` which may cause a further bug when a `@COW` user uses `@COW(storageName:)` to specify a custom storage name. However, the scope of `defaultStorageName` was originally confined in the `COWMacros.swift` with `private` access control. This is a result of I regarded the scope of `defaultStorageName` that it should not be used outside `COWMacros.swift`. When people regarded the scope of all the declarations in the program, then they can add explicit access control to all the parts of the program. Then explicitly marking the access control of the declarations comes to be a small effort with significant increase in the robustness of the code -- since code reviewers can identify declaration abusing by watching the modification to the access control modifier.

I do agree with that enforcing people to write something repeatedly makes no sense. But the enforcement here is "clarity regarding the scope of each declaration of the program" rather than "explicitly writing access control modifier". The access control modifiers should be small efforts when you each time considered the scope of the declarations. If you think these efforts have exceeded your expectations, I'm please to do a favor for you. You've done a good job.

The pre-merge pipeline is a good idea. It would be adopted if there were more people involved in this project.

Reply:
I still think you missed my points, which are:
1. It is perfectly fine for contributors to follow arbitrary coding style/conventions as long as it is enforced in any form (docs, pipeline, etc.). Any violation should be blocked unconditionally. You are the owner of this project and you have the right to do so.
2. It is NOT acceptable to do anything "because the standard library does it". The underlying reason doesn't matter, because its logic just doesn't make sense at the first place.
3. Any discussions/reasoning HERE is a waste of time because of bullet point No.1. Such effort should be made somewhere like [this](https://google.github.io/swift/).